cmake/cares.cmake

```diff
@@ -18,6 +18,10 @@ if("${gRPC_CARES_PROVIDER}" STREQUAL "module")
   endif()
   set(CARES_SHARED OFF CACHE BOOL "disable shared library")
   set(CARES_STATIC ON CACHE BOOL "link cares statically")
+  if(gRPC_BACKWARDS_COMPATIBILITY_MODE)
+    # See https://github.com/grpc/grpc/issues/17255
```

Comment:
can we add a comment here that we expect the C# binary packages to be the only builds that use this?

Reply:
I think the issue link already hints that this is related to C# packaging.